thunder/benchmarks/benchmark_peft.py

```diff
@@ -0,0 +1,724 @@
+
+        executors = thunder.get_default_executors()
+        xforms: list = [NvtxProfileTransform()]
+        logger.info(f"Thunder used executors: {executors}")
+        logger.info(f"Applying Thunder compilation with {len(executors)} executors")
+        be = thunder.dynamo.ThunderCompiler(transforms=xforms, executors=executors)
+        model = torch._dynamo.optimize(backend=be)(model)
+
+    return model
+
+
+def dist_print(*args, **kwargs):
+    """Print only on LOCAL_RANK 0."""
+    if LOCAL_RANK == 0:
+        print(*args, **kwargs)
+
+
+def parse_args():
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--model", default="meta-llama/Llama-3.2-1B")
+    parser.add_argument("--strategy", type=str, default="auto", choices=["auto", "ddp", "fsdp2"])
+    parser.add_argument("--devices", type=int, default=1)
+    parser.add_argument("--max-steps", type=int, default=100)
+    parser.add_argument(
+        "--skip-iters", type=int, default=2, help="Number of warmup iterations to skip in average calculation"
+    )
+    parser.add_argument("--mbs", type=int, default=1)
+    parser.add_argument(
+        "--grad-acc-steps",
+        type=int,
+        default=1,
+        help="Number of steps to accumulate gradients before performing a backward pass.",
+    )
+    parser.add_argument("--seq-length", type=int, default=128)
+    parser.add_argument(
+        "--var-seq-length",
+        action="store_true",
+        help="If true will use different sequence length for each batch; In this case --seq-length will be used as maximum sequence length allowed.",
+    )
+    parser.add_argument(
+        "--jit-backend",
+        default="eager",
+        type=str.lower,
+        choices=["eager", "torchjit", "thunder"],
+    )
+    parser.add_argument("--verbose", action="store_true", help="Enable verbose output including model wrapping details")
+    parser.add_argument("--trust-remote-code", action="store_true")
+    parser.add_argument(
+        "--fixed-num-hidden-layers",
+        type=int,
+        default=None,
+        help="Number of hidden layers to fix the config to.",
+    )
+    parser.add_argument(
+        "--attn-implementation",
+        type=str,
+        default="sdpa",
+        choices=["sdpa", "eager"],
+        help="Attention implementation to use",
+    )
+    parser.add_argument(
+        "--gradient-checkpointing",
+        action="store_true",
+        default=False,
+        help="Enable gradient checkpointing. Disabled by default due to potential compatibility issues with compilers.",
+    )
+
+    args = parser.parse_args()
+
+    # Validate skip-iters
+    if args.skip_iters >= args.max_steps:
+        raise ValueError(f"skip-iters ({args.skip_iters}) must be less than max-steps ({args.max_steps})")
+
+    return args
+
+
+def get_tokenizer(model_name: str, trust_remote_code: bool, fallback_model: str = "gpt2") -> Any:
```

Comment:
Can we refine the type annotation of return value and then remove `from typing import Any`?